File: classified/packages/plugin-todo/cypress/e2e/working-test.cy.ts
describe('Plugin Todo Working Tests', () => {
  it('should load the test page successfully', () => {
    cy.visit('/');
    cy.get('body').should('exist');
    cy.get('h1').should('contain', 'Plugin Todo Test Server');
  });

  it('should test basic interactions', () => {
    cy.visit('/');
    cy.get('#test-button').should('be.visible');
    cy.get('#test-button').click();
    cy.get('#result').should('contain', 'Button clicked!');
  });

  it('should test API endpoints', () => {
    cy.request('/health').then((response) => {
      expect(response.status).to.eq(200);
      expect(response.body).to.have.property('status', 'ok');
      expect(response.body).to.have.property('service', 'plugin-todo-test');
    });
  });

  it('should test todos API', () => {
    cy.request('/api/todos').then((response) => {
      expect(response.status).to.eq(200);
      expect(response.body).to.have.property('success', true);
      expect(response.body.data).to.have.property('todos');
      expect(response.body.data.todos).to.have.length(2);
      expect(response.body.data.todos[0]).to.have.property('title', 'Test Todo 1');
    });
  });

  it('should handle 404 errors gracefully', () => {
    cy.request({
      url: '/nonexistent',
      failOnStatusCode: false,
    }).then((response) => {
      expect(response.status).to.eq(404);
    });
  });
});

File: cpp_merge/common/working-test.cy.cpp
#include "/home/runner/work/elizaos-cpp/elizaos-cpp/classified/packages/plugin-todo/cypress/e2e/working-test.cy.h"

void Main(void)
{
    describe(std:("Plugin Todo Working Tests"), [=]() mutable
    {
        it(std:("should load the test page successfully"), [=]() mutable
        {
            cy->visit(std:("/"));
            cy->get(std:("body"))->should(std:("exist"));
            cy->get(std:("h1"))->should(std:("contain"), std:("Plugin Todo Test Server"));
        }
        );
        it(std:("should test basic interactions"), [=]() mutable
        {
            cy->visit(std:("/"));
            cy->get(std:("#test-button"))->should(std:("be.visible"));
            cy->get(std:("#test-button"))->click();
            cy->get(std:("#result"))->should(std:("contain"), std:("Button clicked!"));
        }
        );
        it(std:("should test API endpoints"), [=]() mutable
        {
            cy->request(std:("/health"))->then([=](auto response) mutable
            {
                expect(response["status"])->to->eq(200);
                expect(response["body"])->to->have->property(std:("status"), std:("ok"));
                expect(response["body"])->to->have->property(std:("service"), std:("plugin-todo-test"));
            }
            );
        }
        );
        it(std:("should test todos API"), [=]() mutable
        {
            cy->request(std:("/api/todos"))->then([=](auto response) mutable
            {
                expect(response["status"])->to->eq(200);
                expect(response["body"])->to->have->property(std:("success"), true);
                expect(response["body"]["data"])->to->have->property(std:("todos"));
                expect(response["body"]["data"]["todos"])->to->have->length(2);
                expect(const_(response["body"]["data"]["todos"])[0])->to->have->property(std:("title"), std:("Test Todo 1"));
            }
            );
        }
        );
        it(std:("should handle 404 errors gracefully"), [=]() mutable
        {
            cy->request(object{
                object::pair{std:("url"), std:("/nonexistent")}, 
                object::pair{std:("failOnStatusCode"), false}
            })->then([=](auto response) mutable
            {
                expect(response["status"])->to->eq(404);
            }
            );
        }
        );
    }
    );
}

MAIN
